app/components/cron-job.js

```diff
@@ -8,6 +8,46 @@ export default Ember.Component.extend({
   actionType: 'Save',
   store: service(),
 
+  intervalText: function() {
```

Comment:
I would split this up into 3 helper methods for extracting the time, calculating the day of the month, and getting the day of the week. Then have the procedural function that is putting together the text variable, call your helper functions. The encapsulation will help test and isolate logic, which is easier to maintain for 3 reasons: 

1. Smaller procedural functions are easier to read and understand for other developers (or yourselves) in the future. 

2. You can unit test the isolated logic (it's easier to cover edge cases of smaller logic).

3. You can replace the logic more easily if needed in the future. For example if we wanted to support another language we could edit the function that grabs weekday names, and add new tests for just that logic without changing the larger procedural function or its integration tests.